LPD-53077 Allow users to easily consume through REST API entries from a root hierarchical structure

ccorreagg opened the pull request:
Related ticket: [LPD-53077](https://liferay.atlassian.net/browse/LPD-53077)

Created from: https://github.com/liferay-bpm/liferay-portal/pull/4570

[LPD-53077]: https://liferay.atlassian.net/browse/LPD-53077?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

carolmariaabb commented on modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/container/request/filter/NestedFieldsContainerRequestFilter.java:
```diff
@@ -0,0 +1,110 @@
+			return;
+		}
+
+		Queue<Long> queue = new LinkedList<>();
+
+		queue.add(_objectDefinition.getObjectDefinitionId());
+
+		Set<Long> processed = new HashSet<>();
+
+		Long objectDefinitionId = null;
+
+		while ((objectDefinitionId = queue.poll()) != null) {
+			if (!processed.add(objectDefinitionId)) {
+				continue;
+			}
+
+			for (ObjectRelationship objectRelationship :
+					_objectRelationshipLocalService.getObjectRelationships(
+						objectDefinitionId, true)) {
+
+				nestedFields.add(objectRelationship.getName());
+
+				if (objectRelationship.getObjectDefinitionId1() !=
+						objectDefinitionId) {
+
+					queue.add(objectRelationship.getObjectDefinitionId1());
+				}
+				else if (objectRelationship.getObjectDefinitionId2() !=
+							objectDefinitionId) {
+
+					queue.add(objectRelationship.getObjectDefinitionId2());
+				}
+			}
+		}
```
Hi @ccorreagg ,

@pedro-oliveira446 updated this logic to use the `ObjectDefinitionTreeFactory` for creating the root model tree, ensuring consistency with [this logic.](https://github.com/carolmariaabb/liferay-portal/blob/2ff83884c136494de528e84602ca2762aa012ca3/modules/apps/object/object-api/src/main/java/com/liferay/object/tree/ObjectDefinitionTreeFactory.java#L50-L65)

Could you please check?

carolmariaabb replied:
you can see the change [here](https://github.com/pedro-oliveira446/liferay-portal/commit/2344d245697eed06082f6d0e7100fefff2467107#diff-36cd0558f894cc2e6e345052868642afebca9f795a1359d1d4c950b007d8d2a3R70-R105)

ccorreagg commented:
Hey @pedro-oliveira446 @carolmariaabb ,

I rebased with the previous PR and added a couple of extra commits with integration tests. Also updated the first commit with Pedro as coauthor.

Could you check please? From my perspective, everything looks ok

ccorreagg commented:
ci:test:sf

liferay-continuous-integration commented:
Test suite: ci:test:relevant has been triggered.

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest(master)/7438/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-34.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7438/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 6 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/fb1c542ccd46c3136eb7c33be1a0c7b9b416db9b">fb1c542ccd46c3136eb7c33be1a0c7b9b416db9b</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/ccorreagg/liferay-portal/tree/LPD-53077-4">LPD-53077-4</a><br/>Branch GIT ID: <a href="https://github.com/ccorreagg/liferay-portal/commit/f058902ddb181722ff478d9bfb32cf61878f047a">f058902ddb181722ff478d9bfb32cf61878f047a</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-34.liferay.com/job/test-portal-source-format/3954/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-34.liferay.com/userContent/jobs/test-portal-source-format/builds/3954/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<h1>Build completed.</h1><p>Testray CSV has been generated successfully for testrayBuildID: <b>228624784</b>.</p><p>Job Link: <a href="https://test-1-6.liferay.com/job/generate-testray-csv/255/">generate-testray-csv</a></p><p>Testray CSV Link: <a href="https://test-1-6.liferay.com/userContent/jobs/generate-testray-csv/builds/255/testray-results-228624784.csv">testray-results-228624784.csv</a></p>

pedro-oliveira446 commented:
ci:forward:force

liferay-continuous-integration commented:
Skipping previously completed test suites:
- `ci:test:relevant`
- `ci:test:sf`
AND Skipping previously passed test suites:
- `ci:test:stable`

liferay-continuous-integration commented:
All required test suite(s) completed.
Forwarding pull request to `brianchandotcom`.
[Console](https://test-1-8.liferay.com/job/forward-pullrequest/798//consoleFull)
